android/src/main/java/com/tailscale/ipn/DnsConfig.java

```diff
@@ -35,31 +38,13 @@
 //
 // --------------------- Android 7 and later -----------------------------------------
 //
-// ## getDnsConfigFromLinkProperties
-// Android provides a getAllNetworks interface in the ConnectivityManager. We walk through
-// each interface to pick the most appropriate one.
-// - If there is an Ethernet interface active we use that.
-// - If Wi-Fi is active we use that.
-// - If LTE is active we use that.
-// - We never use a VPN's DNS servers. That VPN is likely us. Even if not us, Android
-// only allows one VPN at a time so a different VPN's DNS servers won't be available
-// once Tailscale comes up.
-//
-// getAllNetworks() is used as the sole mechanism for retrieving the DNS config with
-// Android 7 and later.
+// ## getDnsConfigs
+// Android provides a requestNetwork interface in the ConnectivityManager. This returns the best
```

Comment:
According to android docs, requestNetwork was added in API level 21, which is Android 5. So, I think we can actually replace all of this, for all versions, with the new code!

Reply:
Yay!